packages/contracts-bedrock/scripts/forge-test-names.ts

```diff
@@ -1,120 +1,86 @@
+    check: (parts: string[]): boolean => {
+      return parts.every((part) => {
+        return part[0] === part[0].toLowerCase()
       })
     },
-    makeError: (name: string): Error =>
-      new Error(
-        `Invalid test name "${name}".\n Test name parts should be in camelCase.`
-      ),
   },
-  invalidNumParts: {
-    check: (name: string, parts: string[]): void => {
-      if (parts.length < 3 || parts.length > 4) {
-        throw FunctionNameChecks.invalidNumParts.makeError(name)
-      }
+  {
+    error:
+      'test names should have either 3 or 4 parts, each separated by underscores',
+    check: (parts: string[]): boolean => {
+      return parts.length === 3 || parts.length === 4
     },
-    makeError: (name: string): Error =>
-      new Error(
-        `Invalid test name "${name}".\n Test names should have either 3 or 4 parts, each separated by underscores.`
-      ),
   },
-  invalidPrefix: {
-    check: (name: string, parts: string[]): void => {
-      if (!['test', 'testFuzz', 'testDiff'].includes(parts[0])) {
-        throw FunctionNameChecks.invalidPrefix.makeError(name)
-      }
+  {
+    error: 'test names should begin with "test", "testFuzz", or "testDiff"',
+    check: (parts: string[]): boolean => {
+      return ['test', 'testFuzz', 'testDiff'].includes(parts[0])
     },
-    makeError: (name: string): Error =>
-      new Error(
-        `Invalid test name "${name}".\n Names should begin with "test", "testFuzz", or "testDiff".`
-      ),
   },
-  invalidTestResult: {
-    check: (name: string, parts: string[]): void => {
-      if (
-        !['succeeds', 'reverts', 'fails', 'benchmark', 'works'].includes(
+  {
+    error:
+      'test names should end with either "succeeds", "reverts", "fails", "works" or "benchmark[_num]"',
+    check: (parts: string[]): boolean => {
+      return (
+        ['succeeds', 'reverts', 'fails', 'benchmark', 'works'].includes(
           parts[parts.length - 1]
-        ) &&
-        parts[parts.length - 2] !== 'benchmark'
-      ) {
-        throw FunctionNameChecks.invalidTestResult.makeError(name)
-      }
+        ) ||
+        (parts[parts.length - 2] === 'benchmark' &&
+          !isNaN(parseInt(parts[parts.length - 1], 10)))
+      )
     },
-    makeError: (name: string): Error =>
-      new Error(
-        `Invalid test name "${name}".\n Test names should end with either "succeeds", "reverts", "fails", "works" or "benchmark[_num]".`
-      ),
   },
-  noFailureReason: {
-    check: (name: string, parts: string[]): void => {
-      if (
-        ['reverts', 'fails'].includes(parts[parts.length - 1]) &&
-        parts.length < 4
-      ) {
-        throw FunctionNameChecks.noFailureReason.makeError(name)
-      }
+  {
+    error:
+      'failure tests should have 4 parts, third part should indicate the reason for failure',
+    check: (parts: string[]): boolean => {
+      return (
+        parts.length !== 4 ||
+        !['reverts', 'fails'].includes(parts[parts.length - 1])
```

Comment:
Fixes a bug and more readable IMO
```suggestion
        !['reverts', 'fails'].includes(parts[parts.length - 1]) ||
        parts.length === 4
```